metrics/meter.go

```diff
@@ -196,55 +196,45 @@ func (m *StandardMeter) Stop() {
 // Count returns the number of events recorded.
 func (m *StandardMeter) Count() int64 {
 	m.lock.RLock()
-	count := m.snapshot.count
-	m.lock.RUnlock()
-	return count
+	defer m.lock.RUnlock()
+	return m.snapshot.count
 }
 
 // Mark records the occurrence of n events.
 func (m *StandardMeter) Mark(n int64) {
-	m.lock.Lock()
-	defer m.lock.Unlock()
-	if m.stopped {
+	stopped := atomic.LoadUint32(&m.stopped)
+	if stopped == 1 {
```

Comment:
In the 'normal path', we're going to do two separate atomic accesses (one load and one add). And in the uncommon path, we're only going to do one. 

Wild idea: how about just blindly doing the add anyway, regardless of whether it's stopped or not? Caveat: I'm assuming we don't stop/start meters, but that the stop is something that, if it does happen, happens during shutdown.

Reply:
I think this is not a bad idea @fjl what do you think?